app/src/org/commcare/CommCareApplication.java

```diff
@@ -1183,4 +1182,8 @@ public AndroidPackageUtils getAndroidPackageUtils() {
     public boolean isNsdServicesEnabled() {
         return true;
     }
+
+    public void initCertificateTransparency() {
+        CommCareNetworkServiceGenerator.customizeRetrofitSetup(new CTInterceptorConfig());

```

Comment:
Hm, are we sure this chains in this way? my read from [the current code](https://github.com/dimagi/commcare-core/blob/1e35c9faaa18743380a30abc7efded5533b941b2/src/main/java/org/commcare/core/network/CommCareNetworkServiceGenerator.java#L116) is that this may be [overwriting this config](https://github.com/dimagi/commcare-android/blob/8a1bf52325ee7198e122eb1e43a21900e33ed29a/app/src/org/commcare/CommCareApplication.java#L255C20-L255C34) rather than chaining it, and we may need a cleaner way to compose these two HttpBuilderConfig classes. It should be easy too, since they both act on the builder, we should just be able to make a composite that runs multiple.

Reply:
@ctsims I had the opportunity to retest this and confirm my understanding. So, this is not supposed to impact any of the existing configuration of the OkHttpClient object such as the [SSLSocketFactory](https://github.com/dimagi/commcare-android/blob/8a1bf52325ee7198e122eb1e43a21900e33ed29a/app/src/org/commcare/network/ISRGCertConfig.java#L74C17-L74C30), it just adds a new network interceptor and it sort of operates at a higher level. For instance, we can set a TrustManager for the Network Interceptor and it shouldn't override the TrustManager of the OkHttpClient. But I do like the idea of having a single entry point for these config classes.